src/pax.rs

```diff
@@ -145,3 +146,74 @@ impl<'entry> PaxExtension<'entry> {
         self.value
     }
 }
+
+/// A builder for PAX extended headers.
+/// It has the data for the extended header
+/// and a status on whether the extended header has been updated.
+pub struct PaxBuilder {
+    data: Vec<u8>,
+    updated: bool,
+}
+
+/// A builder for PAX extended headers.
+impl PaxBuilder {
+    /// Create a new builder for PAX extended headers.
+    pub fn new() -> Self {
+        Self {
+            data: Vec::new(),
+            updated: false,
+        }
+    }
+
+    /// Add a key/value pair to the extended header.
+    ///
+    /// It writes the key/value pair in data Vec to be written to the archive.
+    pub fn add(&mut self, key: &str, value: &str) {
+        self.updated = true;
+        let mut len_len = 1;
+        let mut max_len = 10;
+        let rest_len = 3 + key.len() + value.len();
+        while rest_len + len_len >= max_len {
+            len_len += 1;
+            max_len *= 10;
+        }
+        let len = rest_len + len_len;
+        write!(&mut self.data, "{} {}={}\n", len, key, value).unwrap();
+    }
+
+    /// Getter for the updated field.
+    pub fn updated(&self) -> bool {
+        self.updated
+    }
+
+    /// Getter for the data field as bytes.
+    fn as_bytes(&self) -> &[u8] {
+        &self.data
+    }
+}
+
+/// Extension trait for `Builder` to append PAX extended headers.
+pub trait BuilderExt {
```

Comment:
It looks like this code likely originated outside of this project as an extension trait which makes sense, but I don't see the need for us to have an extension trait for our own structures.